common/src/main/java/org/apache/drill/exec/util/SystemPropertyUtil.java

```diff
@@ -0,0 +1,189 @@
+   * {@code key}, while falling back to {@code null} if the property access fails.
+   *
+   * @return the property value or {@code null}
+   */
+  public static String get(String key) {
+    return get(key, null);
+  }
+
+  /**
+   * Returns the value of the Java system property with the specified
+   * {@code key}, while falling back to the specified default value if
+   * the property access fails.
+   *
+   * @return the property value.
+   *         {@code def} if there's no such property or if an access to the
+   *         specified property is not allowed.
+   */
+  public static String get(final String key, String def) {
+    if (key == null) {
+      throw new NullPointerException("key");
+    }
+    if (key.isEmpty()) {
+      throw new IllegalArgumentException("key must not be empty.");
+    }
+
+    String value = null;
+    try {
+      if (System.getSecurityManager() == null) {
+        value = System.getProperty(key);
+      } else {
+        value = AccessController.doPrivileged(new PrivilegedAction<String>() {
+          @Override
+          public String run() {
+            return System.getProperty(key);
+          }
+        });
+      }
+    } catch (Exception e) {
+        logger.warn("Unable to retrieve a system property '" + key + "'; default values will be used.", e);
+    }
+
+    if (value == null) {
+      return def;
+    }
+
+    return value;
+  }
+
+  /**
+   * Returns the value of the Java system property with the specified
+   * {@code key}, while falling back to the specified default value if
+   * the property access fails.
+   *
+   * @return the property value.
+   *         {@code def} if there's no such property or if an access to the
+   *         specified property is not allowed.
+   */
+  public static boolean getBoolean(String key, boolean def) {
+    String value = get(key);
+    if (value == null) {
+      return def;
+    }
+
+    value = value.trim().toLowerCase();
+    if (value.isEmpty()) {
+      return true;
+    }
+
+    if ("true".equals(value) || "yes".equals(value) || "1".equals(value)) {
+      return true;
+    }
+
+    if ("false".equals(value) || "no".equals(value) || "0".equals(value)) {
+      return false;
+    }
+
+    logger.warn("Unable to parse the boolean system property '{}':{} - using the default value: {}",
+        key, value, def);
+
+    return def;
+  }
+
+  /**
+   * Returns the value of the Java system property with the specified
+   * {@code key}, while falling back to the specified default value if
+   * the property access fails.
+   *
+   * @return the property value.
+   *         {@code def} if there's no such property or if an access to the
+   *         specified property is not allowed.
+   */
+  public static int getInt(String key, int def) {
+    String value = get(key);
+    if (value == null) {
+      return def;
+    }
+
+    value = value.trim().toLowerCase();
+    if (INTEGER_PATTERN.matcher(value).matches()) {
```

Comment:
We don't really need this pattern matcher, if value is not a proper integer `Integer.parseInt` will throw an exception and we already handle that case.